production.py

```diff
@@ -39,6 +48,119 @@ def wkhtml_to_pdf(cls, data, options=None):
             'footer-right': '[page]/[toPage]',
             'footer-spacing': '5',
         }
-        return super(ProductionReport, cls).wkhtml_to_pdf(
+        return super(ReportMixin, cls).wkhtml_to_pdf(
             data, options=options
         )
+
+
+class ProductionReport(ReportMixin):
+    """
+    Production Report
+    """
+    __name__ = 'production.report'
+
+
+class ProductionsReport(ReportMixin):
+    """
+    Productions Report
+    """
+    __name__ = 'report.productions'
+
+    @classmethod
+    def parse(cls, report, records, data, localcontext):
+        Production = Pool().get('production')
+
+        productions = Production.search([
+            ('planned_date', '>=', data['start_date']),
+            ('planned_date', '<=', data['end_date']),
+            ('state', 'not in', ['request', 'cancel'])
+        ])
+        dates = [
+            data['start_date'] + timedelta(days=d) for d in range(
+                (data['end_date'] - data['start_date']).days + 1
+            )
+        ]
+
+        # Dict object to hold productions with key as date and valeue
+        # as list of productions on that day
+        productions_by_date = dict.fromkeys(dates, [])
+
+        # Group searched productions by date and for a dictionary out
+        # of it
+        productions_grouped_by_date = dict(
+            (k, list(v)) for k, v in groupby(
+                sorted(productions, key=lambda p: p.planned_date),
+                key=lambda p: p.planned_date
+            )
+        )
+        # Update the values in productions_by_date
+        productions_by_date.update(productions_grouped_by_date)
+
+        localcontext.update({
+            'productions_by_date': productions_by_date,
+            'dates': dates
+        })
+        return super(ProductionsReport, cls).parse(
+            report, records, data, localcontext
+        )
+
+
+class ProductionsReportWizardStart(ModelView):
+    'Generate Productions Report'
+    __name__ = 'report.productions.wizard.start'
+
+    start_date = fields.Date('Start Date', required=True)
+    end_date = fields.Date('End Date', required=True)
+
+    @staticmethod
+    def default_start_date():
+        """
+        Set default start date to the Monday of current week
+        """
```

Comment:
not sure this is clean enough and works on all cases
https://gist.github.com/sharoonthomas/6e54065bafd57e6a4c51 ?